session/txn.go

```diff
@@ -310,8 +310,8 @@ func keyNeedToLock(k, v []byte) bool {
 	}
 	isDelete := len(v) == 0
 	if isDelete {
-		// only need to delete row key.
-		return k[10] == 'r'
+		// lock row and index key.
+		return k[10] == 'r' || k[10] == 'i'
```

Comment:
It will lock all index keys (unique and non-unique). I think it is OK, but also needs to care about performance.

Reply:
I see, and it seems we can treat delete operations' index locks same to others, PTAL again.